apps/ichat-gw/jabberconnector/JabberComponent.cxx

```diff
@@ -966,25 +968,25 @@ JabberComponent::handlePresence(Stanza *stanza)
 }
 
 void 
-JabberComponent::handleMessage(Stanza* stanza, MessageSession* session)
+JabberComponent::handleMessage(const Message& msg, MessageSession* session)
 {
    std::ostringstream oss;
-   oss << "JabberComponent::handlePresence - " << *stanza;
+   oss << "JabberComponent::handlePresence - " << &msg;
    handleLog(gloox::LogLevelDebug, gloox::LogAreaUser, oss.str());
-   Stanza *s = Stanza::createMessageStanza(stanza->from().full(), "You have reached the ichat gateway!" );
-   s->addAttribute("from", stanza->to().full());
+   Message *s = new Message(Message::Normal, msg.from().full(), "You have reached the ichat gateway!" );
+   s->tag()->addAttribute("from", msg.to().full());
```

Comment:
I wonder if you need to do anything else when constructing a message to avoid that error in prosody.log

Can you find any other examples of code based on gloox and see how they construct a Message?